odl/discr/discr_ops.py

```diff
@@ -499,9 +525,35 @@ def _resize_discr(discr, newshp, offset, discr_kwargs):
         else:
             new_maxpt.append(grid_max[axis] + (num_r + 0.5) * cell_size[axis])
 
-    return uniform_discr(new_minpt, new_maxpt, newshp, dtype=dtype, impl=impl,
-                         exponent=exponent, interp=interp, order=order,
-                         weighting=weighting, nodes_on_bdry=nodes_on_bdry)
+    fspace = FunctionSpace(IntervalProd(new_minpt, new_maxpt),
+                           out_dtype=dtype)
+    dspace = fn(np.prod(newshp), dtype=dtype, impl=impl, exponent=exponent,
+                weighting=weighting)
+
+    # This part is a bit messy since we need to stack together the (old)
+    # nonuniform axes and the (new) uniform axes in the right order
+    nonuni_axes = np.where(np.logical_not(discr.is_uniform_byaxis))[0]
+    if nonuni_axes.size == 0:
+        nonuni_part = []
+    else:
+        nonuni_part = uniform_partition([], [], ())
+        for axis in nonuni_axes:
+            nonuni_part = nonuni_part.append(discr.partition.byaxis[axis])
+
+    part = uniform_partition([], [], ())
+    i_uni = i_nonuni = 0
```

Comment:
Well, I do use them for indexing.. And `num_bla` is for me the *total* number of `bla`. Perhaps `count_bla` instead?